pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -62,15 +63,43 @@ func validateInstanceSpec(spec *sc.InstanceSpec, fldPath *field.Path, create boo
 	return allErrs
 }
 
+func validateInstanceStatus(spec *sc.InstanceStatus, fldPath *field.Path, create bool) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// TODO(vaikas): Implement Status validation.
+	// https://github.com/kubernetes-incubator/service-catalog/issues/882
+	return allErrs
+}
+
+// internalValidateInstanceUpdateAllowed ensures there is not an asynchronous
+// operation ongoing with the instance before allowing an update to go through.
+func internalValidateInstanceUpdateAllowed(new *sc.Instance, old *sc.Instance) field.ErrorList {
+	errors := field.ErrorList{}
+	if old.Status.AsyncOpInProgress {
+		errors = append(errors, field.Forbidden(field.NewPath("Spec"), "Another operation for this service instance is in progress"))
+	}
+	return errors
+}
+
 // ValidateInstanceUpdate validates a change to the Instance's spec.
 func ValidateInstanceUpdate(new *sc.Instance, old *sc.Instance) field.ErrorList {
-	return internalValidateInstance(new, false)
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, internalValidateInstanceUpdateAllowed(new, old)...)
+	allErrs = append(allErrs, internalValidateInstance(new, false)...)
+	return allErrs
+}
+
+func internalValidateInstanceStatusUpdateAllowed(new *sc.Instance, old *sc.Instance) field.ErrorList {
+	errors := field.ErrorList{}
+	// TODO(vaikas): Are there any cases where we do not allow updates to
```

Comment:
To answer your TODO: I think any status update should be allowed with the async operation bit in progress except that you should not be able to update the ready condition to true when 'async op in progress' is true.  IE, if an instance becomes ready due to an async op completing, we should update the async op field in the same status update as we make the ready condition true.